### PR TITLE
Implement basic get() functionality for Spanner rows

### PR DESCRIPTION
**Pull request by dzou**
This implements the get() method for Spanner rows.

Contributes to #34.

### CODE REVIEW

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java [hunk 1]
```diff
@@ -28,19 +32,30 @@
...
    *
    * @param values the list of values in each column.
    * @param rowMetadata the type information for each column.
    */
   public SpannerRow(List<Value> values, StructType rowMetadata) {
     this.values = values;
     this.rowMetadata = rowMetadata;
+
+    this.columnNameIndex = new HashMap<>();
+    for (int i = 0; i < rowMetadata.getFieldsList().size(); i++) {
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]:
```suggestion
    for (int i = 0; i < rowMetadata.getFieldsCount(); i++) {
```

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java [hunk 2]
```diff
@@ -50,7 +65,33 @@ public SpannerRow(List<Value> values, StructType rowMetadata) {
 
   @Override
   public <T> T get(Object identifier, Class<T> type) {
-    // TODO
-    return null;
+
+    int rowIndex;
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]: `columnIndex`?

Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]: Done, renamed.

File: src/test/java/com/google/cloud/spanner/r2dbc/SpannerRowTest.java [hunk 3]
```diff
@@ -18,16 +18,58 @@
...
  */
 public class SpannerRowTest {
 
+  private static final Codecs codecs = new DefaultCodecs();
+
   @Test
-  public void getDummyImplementation() {
-    SpannerRow row = new SpannerRow(null, null);
-    assertThat(row.get("columnName", String.class)).isNull();
+  public void testIndexingIntoColumns() {
```
Comment (meltsufin) [inline src/test/java/com/google/cloud/spanner/r2dbc/SpannerRowTest.java]: Please test exceptions as well.

Comment (dzou) [reply src/test/java/com/google/cloud/spanner/r2dbc/SpannerRowTest.java]: Done.

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java [hunk 4]
```diff
@@ -28,19 +32,30 @@
...
+   * Builds a new Spanner row.
    *
    * @param values the list of values in each column.
    * @param rowMetadata the type information for each column.
    */
   public SpannerRow(List<Value> values, StructType rowMetadata) {
     this.values = values;
     this.rowMetadata = rowMetadata;
+
+    this.columnNameIndex = new HashMap<>();
```
Comment (elefeint) [inline src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]: I wonder if this is something we'll be able to factor out later; no point doing it for each row.

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java [hunk 5]
```diff
@@ -50,7 +65,33 @@ public SpannerRow(List<Value> values, StructType rowMetadata) {
 
   @Override
   public <T> T get(Object identifier, Class<T> type) {
-    // TODO
-    return null;
+
+    int colIndex;
+
+    if (identifier instanceof Integer) {
```
Comment (elefeint) [inline src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]: Maybe move the identifier, uh, identification into a helper method? It's not less complex than `getColIndexByName()`.

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java (hunk 4 above)
Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]: Good point. I might be taking on this when I implement the SpannerRowMetadata object; perhaps this index could be stored there...

Perhaps every SpannerRow could have a reference to a SpannerRowMetadata instance?

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java (hunk 5 above)
Comment (dzou) [reply src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]: Done.

File: src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java (hunk 4 above)
Comment (elefeint) [reply src/main/java/com/google/cloud/spanner/r2dbc/SpannerRow.java]: Yeah, I was thinking along these lines.

**Pull request merged**